src/System.Linq.Expressions/src/System/Linq/Expressions/NewArrayExpression.cs

```diff
@@ -63,15 +63,20 @@ protected internal override Expression Accept(ExpressionVisitor visitor)
         /// <returns>This expression if no children changed, or an expression with the updated children.</returns>
         public NewArrayExpression Update(IEnumerable<Expression> expressions)
         {
-            if (expressions == Expressions)
+            // Explicit null check here as otherwise wrong parameter name will be used.
+            ContractUtils.RequiresNotNull(expressions, nameof(expressions));
```

Comment:
This fixes a minor bug in the existing implementation, which would have either `"initializers"` or `"bounds"` as the parameter name when null was passed.